### PR TITLE
Use Path for reflect.json access in GraalTypeElementVisitor

### PR DESCRIPTION
**Pull request by yawkat**
Transforming to File is unnecessary here, and may be leading to #7257

### CODE REVIEW

File: graal/src/main/java/io/micronaut/graal/reflect/GraalTypeElementVisitor.java [hunk 1]
```diff
@@ -302,19 +302,19 @@ public final void finish(VisitorContext visitorContext) {
...
-        File userReflectJsonFile = projectDir
-                .map(projectPath -> Paths.get(projectPath.toString(), BASE_REFLECT_JSON).toFile())
+        Path userReflectJsonFile = projectDir
+                .map(projectPath -> projectPath.resolve(BASE_REFLECT_JSON))
                 .orElse(null);
 
-        if (userReflectJsonFile != null && userReflectJsonFile.exists()) {
-            try {
-                json = MAPPER.readValue(userReflectJsonFile, new TypeReference<List<Map>>() {
+        if (userReflectJsonFile != null && Files.exists(userReflectJsonFile)) {
```
Comment (timyates) [inline graal/src/main/java/io/micronaut/graal/reflect/GraalTypeElementVisitor.java:313]: Do we have anything that tests this?  Any other projects?  I don't think this has changed the coverage, as it probably wasn't tested before.

**Review by timyates — APPROVED** [review]
Not sure how this fixes the original issue (as the commit they made to Turbine seems a much more likely candidate), but I tend to prefer Path over File, so lgtm 👍

**Comment by yawkat** [comment]
@timyates if I understand correctly, some sort of VFS is involved, which would break if using File. but this is pure speculation, it just seems like the most likely place where our code is sensitive to an invalid URI.

File: graal/src/main/java/io/micronaut/graal/reflect/GraalTypeElementVisitor.java (hunk 1 above)
Comment (yawkat) [reply graal/src/main/java/io/micronaut/graal/reflect/GraalTypeElementVisitor.java:313]: i dont know a way to build a test that would make using path only here behave any differently, unfortunately.

**Review by graemerocher — COMMENTED** [review]
One note about this fix is this code has been completely re-written in 3.5.x so we may have merge challenges when merging up.

**Comment by yawkat** [comment]
Oh right, it generates a Feature now doesnt it? in that case it's easier to just wait for that change

**Pull request closed without merging**